spring-integration-core/src/main/java/org/springframework/integration/store/MessageGroupStore.java

```diff
@@ -140,14 +129,31 @@
 	MessageGroupMetadata getGroupMetadata(Object groupId);
 
 	/**
-	 * Return the one {@link org.springframework.messaging.Message} from {@link org.springframework.integration.store.MessageGroup}.
+	 * Return the one {@link Message} from {@link MessageGroup}.
 	 * @param groupId The group identifier.
-	 * @return the {@link org.springframework.messaging.Message}.
+	 * @return the {@link Message}.
 	 * @since 4.0
 	 */
 	Message<?> getOneMessageFromGroup(Object groupId);
 
 	/**
+	 * Store messages with an association to a group id.
+	 * This can be used to group messages together.
+	 * @param groupId The group id to store messages under.
+	 * @param messages The messages to add.
+	 * @since 4.3
+	 */
+	void addMessagesToGroup(Object groupId, Message<?>... messages);
```

Comment:
What is the purpose of this? I don't see it called with more than one message and it adds overhead in that case.